File: Assignment_14/program1.c
 ////////////////////////////////////////////////////////////////////////////////////////////
//Author : Prabhanjan Sanjay Ingle
//Description : Print the Pattern from given input rows and coloumns
//Input : rows : 4 , Coloumns : 4
//1 2 3 4
//5 6 7 8
//9 1 2 3
//4 5 6 7
//Date : 03-06-2025
////////////////////////////////////////////////////////////////////////////////////////////
#include <stdio.h>

void Pattern(int iRow, int iCol)
{
    int iCnt1 = 0,iCnt2 = 0,iCnt3 = 1;
 
   
    for(iCnt1 = 1; iCnt1 <= iRow; iCnt1++)
    {
        
       for(iCnt2 = 1; iCnt2 <= iCol; iCnt2++ )
       {
        if(iCnt3 < 10 && iCnt3 >= 1)
        {
            printf("%d\t",iCnt3);
            iCnt3++;
        }
        else 
        {
            iCnt3 = 1;
            printf("%d\t",iCnt3);
            iCnt3++;
            
        }    
           

       }
        printf("\n");
    }
}


int main()
{
    int iValue1 = 0;
    int iValue2 = 0;

    printf("Enter rows  : \n");
    scanf("%d", &iValue1);
    printf("Enter coloumns : \n");
    scanf("%d", &iValue2);

    Pattern(iValue1,iValue2);

    return 0;



}

File: Assignment_44/program1.java
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//Author : Prabhanjan Sanjay Ingle
//Description : Given the 2d matrix return the value of addition of its diagonals 
/*
    Input :123
           456
           789
           
    Output :29
*/
//Date : 29-07-2025
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////



import java.util.*;

class matrix
{
    private int Arr[][];

    public matrix(int A,int B)
    {
        Arr = new int[A][B];
    }

    public void Accept()
    {
        System.out.println("please enter the elements of matrix : ");

        Scanner sobj = new Scanner(System.in);
        int i = 0, j = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length;j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
        }

    }
    public int AdditionDiagonals()
    {
        int i = 0, j = 0,iSum = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for( j = 0; j < Arr[i].length;j++)
            {
                if(Arr[i] == Arr[j])
                {
                    iSum = iSum + Arr[i][j];

                }
            }
        }

        return iSum;
    }

    
    
}

public class program1
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        
        System.out.println("Enter the number of rows : ");
        int iRow = sobj.nextInt();

        System.out.println("Enter the number of coloumns : ");
        int iCol = sobj.nextInt();

        
        matrix mobj = new matrix(iRow, iCol);

        mobj.Accept();

        int iRet = mobj.AdditionDiagonals();
        System.out.println("Addition of diagonals of given matrix is : " + iRet);       
    }
    
}
